File: test/resources/HelloProphecy/prophecy/pipelines/gold_top_customers/code/gold_top_customers/udfs/UDFs.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.lookups import (
    createLookup,
    createRangeLookup,
    lookup,
    lookup_last,
    lookup_match,
    lookup_count,
    lookup_row,
    lookup_row_reverse,
    lookup_nth
)

def registerUDFs(spark: SparkSession):
    spark.udf.register("rand_zip_index", rand_zip_index)
    

    try:
        from prophecy.utils import ScalaUtil
        ScalaUtil.initializeUDFs(spark)
    except :
        pass

@udf(returnType = IntegerType())
def rand_zip_index(value: int):
    from random import Random
    r = Random(value)

    return r.randrange(0, 27689)

File: test/resources/HelloProphecy/prophecy_scala/pipelines/raw_bronze/code/src/main/scala/io/prophecy/pipelines/raw_bronze/udfs/UDFs.scala
package io.prophecy.pipelines.raw_bronze.udfs

import _root_.io.prophecy.abinitio.ScalaFunctions._
import _root_.io.prophecy.libs._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql._

object UDFs extends Serializable {

  def registerUDFs(spark: SparkSession) = {
    spark.udf.register("rand_zip_index", rand_zip_index)
    try registerAllUDFs(spark)
    catch {
      case _ => ()
    }
  }

  def rand_zip_index = {
    udf { (value: Int) =>
      import scala.util.Random
      val r = new Random(value)
      r.nextInt(27689)
    }
  }

}

object PipelineInitCode extends Serializable
